### DISCUSSION TITLE
CUDA acceleration

### DISCUSSION THREAD

**Discussion by emeQee** [microsoft/AirSim#4229] [Support Q&A]
I have written a python script to receive uncompressed float depth images in a while loop. The FPS I receive is around 0.085 fps and not suitable for live my use case.

My question is whether AirSim uses the CUDA enabled GPU by default or are there more steps involved to do so? How can I accelerate the requests?

My System:
- Intel i7-2600K @ 3.4 GHz
- RAM 16 GB
- Win10 Pro x64 (19042.1348)
- nVidia GeForce GTX 1060 6GB
- AirSim v1.6.0windows
- CUDA 10.5 and Toolkits installed
- Run on packaged project

```
while True:
  responses = self.client.simGetImages([
    airsim.ImageRequest(self.camera_name, airsim.ImageType.DepthPerspective, True, False)
  ])
```

**Comment by zimmy87** [comment]
From doing a search of the project, it doesn't look like we're currently using CUDA for anything. I suspect enabling this will require some feature work.
